### PR TITLE
Add hermetic test running as non-root user

### PR DESCRIPTION
**Pull request by priyawadhwa**
Add a hermekton test running as a non-root user. ref https://github.com/tektoncd/pipeline/issues/3965

# Changes

/kind misc
/release-note-none

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Release notes block below has been filled in or deleted (only if no user facing changes)

```release-note
NONE
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,6 +3,7 @@ module github.com/tektoncd/pipeline
 go 1.13
 
 require (
+	github.com/AlekSi/pointer v1.1.0 // indirect
```
Comment (sbwsg) [inline go.mod:6]: It looks like the test is using [`k8s.io/utils/pointer`](https://github.com/kubernetes/utils/blob/da695404a2bc/pointer/pointer.go#L114).  I'm not 100% sure but it doesn't look like this module is used.

**Review by sbwsg — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (sbwsg) [reply go.mod:6]: The code for the test file [is importing `k8s.io/utils/pointer`](https://github.com/tektoncd/pipeline/pull/3973/files#diff-2b91646885f5ea26b953eaab2bc73293cb29ba32a0de26c46d3f2521f558d8a5R30) for those funcs I think, isn't it?  I might be totally misunderstanding what's happening.

Comment (priyawadhwa) [reply go.mod]: Oh I didn't even see that ! Thanks for pointing it out, fixed 😄

File: test/hermetic_taskrun_test.go [hunk 2]
```diff
@@ -91,3 +110,35 @@ apt-get install -y curl`,
...
+		},
+		Spec: v1beta1.TaskRunSpec{
+			Timeout: &metav1.Duration{Duration: time.Minute},
+			TaskSpec: &v1beta1.TaskSpec{
+				Steps: []v1beta1.Step{
+					{
+						Container: corev1.Container{
+							Name:  "curl",
+							Image: "gcr.io/cloud-builders/curl",
+							SecurityContext: &corev1.SecurityContext{
```
Comment (pritidesai) [inline test/hermetic_taskrun_test.go:130]: thanks @priyawadhwa for the changes, can this `SecurityContext` be configured for a pod?

**Comment by pritidesai** [comment]
very exciting, thanks @priyawadhwa 👍 

/lgtm

File: test/hermetic_taskrun_test.go (hunk 2 above)
Comment (pritidesai) [reply test/hermetic_taskrun_test.go:130]: we can include an additional test with security context at the pod level ...

**Pull request merged**